src/daemon/database/interface.rs

```diff
@@ -855,3 +855,51 @@ pub fn db_vaults_from_spend(
 
     Ok(db_vaults)
 }
+
```

Comment:
FIXME: how do I merge db_transaction_has_priority and db_transaction_has_priority_dbtx? 😭

Reply:
You can't without an invasive refactoring. You could still do that to DRY the more footguny part:
```diff
diff --git a/src/daemon/database/interface.rs b/src/daemon/database/interface.rs
index 9e03856..d4f36a0 100644
--- a/src/daemon/database/interface.rs
+++ b/src/daemon/database/interface.rs
@@ -86,6 +86,7 @@ where
     x
 }
 
+// FIXME: have a version mergeable with the "regular" db_query
 fn db_query_tx<'a, P, F, T>(
     db_tx: &Transaction,
     stmt_str: &'a str,
@@ -97,7 +98,6 @@ where
     P::Item: ToSql,
     F: FnMut(&Row<'_>) -> rusqlite::Result<T>,
 {
-    // rustc says 'borrowed value does not live long enough'
     db_tx
         .prepare(stmt_str)
         .map_err(|e| DatabaseError(format!("Preparing query: '{}'", e.to_string())))?
@@ -856,18 +856,21 @@ pub fn db_vaults_from_spend(
     Ok(db_vaults)
 }
 
-pub fn db_transaction_has_priority(db_path: &Path, txid: &Txid) -> Result<bool, DatabaseError> {
-    // We don't know if this tx is an unvault or a spend...
-    db_query(
-        db_path,
-        "SELECT has_priority \
-         FROM spend_transactions as stx \
-         INNER JOIN spend_inputs as sin ON stx.id = sin.spend_id \
-         INNER JOIN presigned_transactions as ptx ON ptx.id = sin.unvault_id \
-         WHERE stx.txid = (?1) OR ptx.txid = (?1)",
-        params![txid.to_vec()],
-        |row| Ok(row.get::<_, Option<bool>>(0)? == Some(true)),
-    )
+const TX_HAS_PRIO_QUERY: &str = "\
+SELECT has_priority
+    FROM spend_transactions as stx
+        INNER JOIN spend_inputs as sin ON stx.id = sin.spend_id
+        INNER JOIN presigned_transactions as ptx ON ptx.id = sin.unvault_id
+    WHERE stx.txid = (?1) OR ptx.txid = (?1)
+";
+
+pub fn db_transaction_has_priority_dbtx(
+    db_tx: &Transaction,
+    txid: &Txid,
+) -> Result<bool, DatabaseError> {
+    db_query_tx(db_tx, TX_HAS_PRIO_QUERY, params![txid.to_vec()], |row| {
+        Ok(row.get::<_, Option<bool>>(0)? == Some(true))
+    })
     .map(
         // if the tx is not here, no need to return an error: it doesn't have priority
         |mut v| {
@@ -879,20 +882,11 @@ pub fn db_transaction_has_priority(db_path: &Path, txid: &Txid) -> Result<bool,
     )
 }
 
-pub fn db_transaction_has_priority_dbtx(
-    db_tx: &Transaction,
-    txid: &Txid,
-) -> Result<bool, DatabaseError> {
-    db_query_tx(
-        db_tx,
-        "SELECT has_priority \
-         FROM spend_transactions as stx \
-         INNER JOIN spend_inputs as sin ON stx.id = sin.spend_id \
-         INNER JOIN presigned_transactions as ptx ON ptx.id = sin.unvault_id \
-         WHERE stx.txid = (?1) OR ptx.txid = (?1)",
-        params![txid.to_vec()],
-        |row| Ok(row.get::<_, Option<bool>>(0)? == Some(true)),
-    )
+pub fn db_transaction_has_priority(db_path: &Path, txid: &Txid) -> Result<bool, DatabaseError> {
+    // We don't know if this tx is an unvault or a spend...
+    db_query(db_path, TX_HAS_PRIO_QUERY, params![txid.to_vec()], |row| {
+        Ok(row.get::<_, Option<bool>>(0)? == Some(true))
+    })
     .map(
         // if the tx is not here, no need to return an error: it doesn't have priority
         |mut v| {
```